Beams/MRML/vtkMRMLRTBeamNode.cxx

```diff
@@ -916,3 +924,138 @@ void vtkMRMLRTBeamNode::CreateMLCPointsFromSectionBorder(double jawBegin,
   }
   side12.push_back(side2.back());
 }
+
+void vtkMRMLRTBeamNode::SetDoseInfluenceMatrixFromTriplets(
+    int numRows, int numCols,
+    DoseInfluenceMatrixIndexVector& rows,
+    DoseInfluenceMatrixIndexVector& columns,
+    DoseInfluenceMatrixValueVector& values,
+    double* doseGridDim,
+    double* doseGridSpacing
+)
+{
+  typedef Eigen::Triplet<double> T;
+  std::vector<T> tripletList;
+  tripletList.reserve(values.size());
+  
+  for (size_t i = 0; i < values.size(); ++i)
+  {
+    tripletList.push_back(T(rows[i], columns[i], values[i]));
+  }
+
+  this->DoseInfluenceMatrix = DoseInfluenceMatrixType(numRows, numCols);
+  this->DoseInfluenceMatrix.setFromTriplets(tripletList.begin(), tripletList.end());
+
+  // Store dose grid dimensions and spacing on which the dose influence matrix is defined
+  for (int i = 0; i < 3; ++i)
+  {
+	  if (doseGridDim == nullptr)
+	  {
+		  this->DoseGridDim[i] = -1;
+	  }
+      else
+	  {
+		  this->DoseGridDim[i] = doseGridDim[i];
+	  }
+
+	  if (doseGridSpacing == nullptr)
+	  {
+		  this->DoseGridSpacing[i] = -1;
+	  }
+      else
+      {
+          this->DoseGridSpacing[i] = doseGridSpacing[i];
+      }
+  }
+
+}
+
+//---------------------------------------------------------------------------
+int vtkMRMLRTBeamNode::GetDoseInfluenceMatrixRows()
+{
+  return this->DoseInfluenceMatrix.rows();
+}
+
+//---------------------------------------------------------------------------
+int vtkMRMLRTBeamNode::GetDoseInfluenceMatrixColumns()

```

Comment:
From the header I can see that this is for getting the number of rows, but the function name is a bit ambiguous. I'd use `GetDoseInfluenceMatrixColumnCount`, or `GetDoseInfluenceMatrixNumberOfColumns`. Same with the other one for rows.